### DISCUSSION TITLE
Help: Oops! We could not find requested route.

### DISCUSSION THREAD

**Discussion by Caesarovich** [shlinkio/shlink#1946] [Help wanted] [answered]
### Shlink version

v3.7.0

### PHP version

container

### How do you serve Shlink

RoadRunner Docker image

### Database engine

SQLite

### Database version

0.0.0

### Summary

Hi, I'm trying to setup shlink with Docker compose like so:

```yml
    restarted: true
    project_name: shlink
    pull: true
    definition:
      version: '3'
      services:
        server:
          restart: unless-stopped
          image: shlinkio/shlink:stable
          volumes:
            - '/etc/shlink/data'
          environment:
            DEFAULT_DOMAIN: 's.XXX'
            IS_HTTPS_ENABLED: 1
          ports:
            - 'XXXX:8080'
```

I can connect to the web interface at s.XXXX, I create a shortened URL but when I click it the following error appears:

> Oops! We could not find requested route.
>
> Use your browser's back button to navigate to the page you have previously come from, or just press this button.

I'm not sure what the issue is but I think it's trying to find the shortened URL on the web client instead of redirecting to its original URL ?

**Comment by Caesarovich** [comment]
Follow up: I found out that the shortened URL works when using another device / private mode on my browser. Which means the URL redirections don't work when 'logged in' to the web interface. I'm not sure if this should be converted into an issue or not.

**Comment by acelaya** [comment, marked as answer]
The issue here is most probably that you didn't set the proper value in `DEFAULT_DOMAIN`.

### TL;DR

Shlink and shlink-web-client are effectively two independent apps.

Shlink does not use `DEFAULT_DOMAIN` when processing incoming requests. Any request that gets to it will be handled, even if it was for a different domain.

The reason for that value to exist is to identify known non-default domains, and to properly build short URLs when served via API.

shlink-web-client fetches data from Shlink via REST API. Short URLs displayed in shlink-web-client are built by Shlink using `DEFAULT_DOMAIN`.

When you are in the browser, and a page creates a link, the browser will determine if that URL belongs to the loaded app itself, or is an external domain.

### Potential reasons for this behavior

1. You are trying to serve both Shlink and shlink-web-client in the same URL, without different paths. That won't work, and [you need to specify a different path for one of them](https://shlink.io/documentation/advanced/shlink-architecture/#why-cant-i-serve-shlink-and-a-ui-like-shlink-web-client-from-the-same-domain) so that routing can properly work
2. You did not set `DEFAULT_DOMAIN` at all, which will cause Shlink to serve short URLs that could be identified by the browser as relative URLs.
3. If the value in `DEFAULT_DOMAIN` happens to incorrectly match the URL of shlink-web-client, the browser will try to open them "locally".
